src/coreclr/jit/emitxarch.cpp

```diff
@@ -13672,7 +13684,8 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
         case IF_RWR_LABEL:
         case IF_SWR_LABEL:
             assert(id->idGCref() == GCT_NONE);
-            assert(id->idIsBound());
+            assert(id->idIsBound() ||
```

Comment:
Can you move `(emitIsUncondJump(id) && id->idCodeSize() == 0 && ((instrDescJmp*)id)->idjIsJmpAlways == 1)` inside `emitInstHasNoCode()`?

Reply:
I can't move it because the sig is `bool emitInstHasNoCode(instruction ins)` but I could create a second version which takes a `instrDesc` parameter and checks for align and then the jump checks, do you want me to do that?

Reply:
Ok, I hadn't checked the number of call sites but it's only two so I've made this change.